Adding example for S3 Express use case (#92)
awslabs/mountpoint-s3-csi-driver: examples/kubernetes/static_provisioning/README.md

## Examples in this folder
- `static_provisioning.yaml` - spawning a pod which creates a file with name as the current date/time
- `non_root.yaml` - same as above, but the pod is spawned as non-root (uid `1000`, gid `2000`)

> - `s3_express_specify_az.yaml` - same as above, but this uses [S3 Express](https://docs.aws.amazon.com/AmazonS3/latest/userguide/s3-express-one-zone.html) bucket and shows how to specify the availability zone (AZ) of the [pod assignment](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity)

Reviewer: IMO it's better to be explicit and use the full "S3 Express One Zone" name in this doc even if it is a bit long.

Reviewer 2: yeah, it's always "S3 Express One Zone"

Reviewer 2: ```suggestion
- `s3_express_specify_az.yaml` - same as above, but this uses a [S3 Express One Zone](https://docs.aws.amazon.com/AmazonS3/latest/userguide/s3-express-one-zone.html) directory bucket and shows how to specify the availability zone (AZ) of the [pod assignment](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity) to co-locate the pod with the bucket for lower latency access
```

## Configure
### Edit [Persistent Volume](https://github.com/awslabs/mountpoint-s3-csi-driver/blob/main/examples/kubernetes/static_provisioning/static_provisioning.yaml)

> > Note: This example assumes your S3 bucket has already been created. If you need to create a bucket, follow the [S3 documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-bucket.html) for a regular bucket and [S3 One Zone documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/directory-bucket-create.html) for an S3 Express bucket (aka a Directory bucket).

Reviewer 3: This says S3 One Zone, in line with the above comment it should probably be S3 Express One Zone

Author: fixed in latest commit

> > Note: This example assumes your S3 bucket has already been created. If you need to create a bucket, follow the [S3 documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-bucket.html) for a regular bucket and [S3 One Zone documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/directory-bucket-create.html) for an S3 Express bucket (aka a Directory bucket).

Reviewer 2: ```suggestion
> Note: This example assumes your S3 bucket has already been created. If you need to create a bucket, follow the [S3 documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-bucket.html) for a general purpose bucket or the [S3 Express One Zone documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/directory-bucket-create.html) for a directory bucket.
```
